packages/react-router/index.tsx

```diff
@@ -1032,14 +1035,18 @@ function safelyDecodeURIComponent(value: string, paramName: string) {
  *
  * @see https://reactrouter.com/api/resolvePath
  */
-export function resolvePath(to: To, fromPathname = '/'): Path {
+export function resolvePath(to: To, fromPathname = '/', basename = ''): Path {
   let { pathname: toPathname, search = '', hash = '' } =
     typeof to === 'string' ? parsePath(to) : to;
 
   let pathname = toPathname
     ? resolvePathname(
         toPathname,
-        toPathname.startsWith('/') ? '/' : fromPathname
+        toPathname.startsWith('/')
+          ? basename
+            ? `/${basename}`
```

Comment:
Since `basename` is coming through the public API we should be sure to strip any preceeding slash before blindly adding one more here. Otherwise this PR looks great 👍